### PR TITLE
Refactor Formatted and WideString

### PR DESCRIPTION
**Pull request by Xitian9**
This implements some simple refactors to make functions do less work.

### CODE REVIEW

File: test-suite/TestSpec.hs [hunk 1]
```diff
@@ -283,12 +283,12 @@ spec = do
...
             it "drops 3" $ buildCell (dropLeft 3 exampleF) `shouldBe` "XXXloZthereWYYY"
             it "drops 8" $ buildCell (dropLeft 8 exampleF) `shouldBe` "XXXZreWYYY"
-            it "drops 20" $ buildCell (dropLeft 20 exampleF) `shouldBe` "XXXZWYYY"
+            it "drops 20" $ buildCell (dropLeft 20 exampleF) `shouldBe` ""
         describe "dropRight" $ do
             it "drops 0" $ buildCell (dropRight 0 exampleF) `shouldBe` "XXXHelloZthereWYYY"
             it "drops 3" $ buildCell (dropRight 3 exampleF) `shouldBe` "XXXHelloZthWYYY"
-            it "drops 8" $ buildCell (dropRight 8 exampleF) `shouldBe` "XXXHeZWYYY"
-            it "drops 20" $ buildCell (dropRight 20 exampleF) `shouldBe` "XXXZWYYY"
+            it "drops 8" $ buildCell (dropRight 8 exampleF) `shouldBe` "XXXHeYYY"
```
Comment (muesli4) [inline test-suite/TestSpec.hs]: Is this really desired behavior?  We expect formatting instructions to not be visible from a length perspective.  Could there be a situation where it is desired to still have the formatting instructions?

File: src/Text/Layout/Table/Cell/WideString.hs [hunk 2]
```diff
@@ -21,26 +21,44 @@ instance Cell WideString where
...
   where
     charLen = charWidth x
 
+-- | Drop characters from the left side of a 'String', recording how much padding
+-- is needed to have dropped the requested width.
+trimLeft :: Int -> String -> CellView String
+trimLeft = dropWide True
+
+-- | Drop characters from the right side of a 'String', recording how much padding
+-- is needed to have dropped the requested width.
```
Comment (muesli4) [inline src/Text/Layout/Table/Cell/WideString.hs]: This sentence sounds weird to me.  Maybe it is just me.

**Review by muesli4 — COMMENTED** [review]
Do the tests pass?

File: test-suite/TestSpec.hs (hunk 1 above)
Comment (Xitian9) [reply test-suite/TestSpec.hs]: It's a good question, and one I've gone back and forth on. I'm happy for you to make the call that the formatting should remain even with empty content.

Comment (Xitian9) [reply test-suite/TestSpec.hs]: I've changed it so it leaves formatting around empty elements alone. That's certainly the ‘least surprise’ behaviour, and avoids trying to be too clever.

File: src/Text/Layout/Table/Cell/WideString.hs (hunk 2 above)
Comment (Xitian9) [reply src/Text/Layout/Table/Cell/WideString.hs]: Altered the wording.

**Comment by Xitian9** [comment]
Actually, I'm going to mark this draft for now. I suspect this will all be redundant after implementing the other proposed changes.

**Comment by Xitian9** [comment]
I'll go ahead and remove this entirely to reduce PR clutter. It is outdated.

**Pull request closed without merging**